### PR TITLE
 Trade Ship animated with a new sprite

### PR DESCRIPTION
**Pull request by Tranberry**
@Raketfart help me!

12basichaulship

### CODE REVIEW

**Comment by Tranberry** [comment]
screencap

**Comment by Raketfart** [comment]
haha - excellent gif

**Comment by Tranberry** [comment]
I was in a hurry :dango:

**Comment by sboigelot** [comment]
Did you take into account that the sprite name of the ship is defined in the trader prototype, "basic haul ship" will not be a static name and shoudn't be use as it in code

File: Assets/Scripts/Controllers/TradeController.cs [hunk 1]
```diff
@@ -63,6 +64,8 @@ public void CallTradeShipTest(Furniture landingPad)
         SpriteRenderer spriteRenderer = go.AddComponent<SpriteRenderer>();
         spriteRenderer.sprite = SpriteManager.GetSprite("Trader", "BasicHaulShip");
         spriteRenderer.sortingLayerName = "TradeShip";
+        controller.Animation = new SpritenameAnimation("flying", new string[] { "BasicHaulShip_01", "BasicHaulShip_02" }, 1 / 4);
```
Comment (sboigelot) [inline Assets/Scripts/Controllers/TradeController.cs]: this shouldn't be a static string array, but is dependent of the trader prototype

Comment (Tranberry) [inline Assets/Scripts/Controllers/TradeController.cs]: @Raketfart you where right, I'll have to look at your code. :( But I might learn something.

**Comment by Raketfart** [comment]
Ah sorry about the hardcoded string. It was made a too fast I think. I can take a look tonight

**Comment by sboigelot** [comment]
@Raketfart no rush :) take your time

**Comment by Raketfart** [comment]
@sboigelot Hmmm... At first glance, I can't find the sprite name in the prototype.
I see the controller calling `spriteRenderer.sprite = SpriteManager.GetSprite("Trader", "BasicHaulShip");`

**Comment by Raketfart** [comment]
I didn't see the anything exsisting code in the trader prototype. But since the tradeship has no xml, I added the animation to the trader xml. Then the animations are copied to the tradeship when creating a new trader.

**Comment by sboigelot** [comment]
Ok oups then it's TODO that hasn't been done yet.  @Raketfart seems great, testing it soon

**Comment by sboigelot** [comment]
The second trader prototype in trader.xml do not have an animation defined, will this cause an issue?

**Comment by Raketfart** [comment]
Good question. I don't know if it is used, but I'll add animation just in case

**Comment by Raketfart** [comment]
2nd trader also has the same animation now.

**Pull request merged**